TWLight/users/management/commands/user_renewal_notice.py

```diff
@@ -13,13 +17,32 @@ class Command(BaseCommand):
     def handle(self, *args, **options):
         # Get all authorization objects with an expiry date in the next
         # four weeks, for which we haven't yet sent a reminder email, and
-        # exclude users who disabled these emails.
-        expiring_authorizations = Authorization.objects.filter(
-            date_expires__lt=datetime.today() + timedelta(weeks=2),
-            date_expires__gte=datetime.today(),
-            reminder_email_sent=False,
-            partners__isnull=False,
-        ).exclude(user__userprofile__send_renewal_notices=False)
+        # exclude users who disabled these emails and who have already filed
+        # for a renewal.
+        editor_qs = Editor.objects.select_related("user")
+        users_with_applications_for_renewal = (
```

Comment:
Just quickly looking at this so sorry if I've missed something - we specifically want to discount users who have a renewal request for the resource to which their authorization is expiring. Is this queryset editors with any renewal request or specifically for this resource?

Reply:
Great question. I will create a test case where a user has 2 soon-to-expire auths and has just renewed one. They should only get one email. That will test if this code is working correctly :)

Reply:
You were absolutely right: the code as it was didn't work as expected. Have changed it and added new test cases.